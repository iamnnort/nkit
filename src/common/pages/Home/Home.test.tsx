import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import HomeComponent from './Home';
import { TestProvider } from '../../../../test';

describe('HomeComponent', () => {
  let wrapper: ReactWrapper;

  const store = {
    git: {
      isInitialLoaded: true,
      isLoading: false,
      error: '',
      releases: [
        {
          html_url: 'https://github.com/nikitapavets/react-ssr-ts-redux/releases/tag/v1',
          id: 1,
          name: 'Release v1',
          tag_name: 'v1',
        },
      ],
    },
  };

  beforeEach(() => {
    wrapper = mount(
      <TestProvider store={store}>
        <HomeComponent.component />
      </TestProvider>
    );
  });

  it(`should not regret`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
