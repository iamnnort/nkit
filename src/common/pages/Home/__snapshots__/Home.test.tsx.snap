// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should not regret 1`] = `
.c0 {
  margin-top: 100px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  max-width: 500px;
  margin: auto;
  padding: 30px;
  background: #fff;
}

.c2 {
  font-size: 24px;
  font-weight: bold;
  text-align: center;
}

.c3 {
  margin-top: 15px;
}

.c4 {
  margin-top: 15px;
}

<Component
  store={
    Object {
      "git": Object {
        "error": "",
        "isInitialLoaded": true,
        "isLoading": false,
        "releases": Array [
          Object {
            "html_url": "https://github.com/nikitapavets/nkit/releases/tag/v1",
            "id": 1,
            "name": "Release v1",
            "tag_name": "v1",
          },
        ],
      },
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "brandColor": "#6B71D9",
          "errorColor": "#f00",
          "largeFontSize": "24px",
          "mediumFontSize": "18px",
          "mediumPadding": "30px",
          "shadowColor": "rgba(0, 0, 0, 0.5)",
          "smallFontSize": "14px",
          "smallPadding": "15px",
          "textColor": "#444",
          "textFontFamily": "\\"Roboto\\", sans-serif",
          "textFontSize": "14px",
          "whiteColor": "#fff",
        }
      }
    >
      <HomeComponent>
        <styled.section>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "bDWFJH",
                  "rules": Array [],
                },
                "displayName": "styled.section",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "section",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <section
              className=""
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bwzfXH",
                        "isStatic": true,
                        "lastClassName": "c0",
                        "rules": Array [
                          "
  margin-top: 100px;

  display: flex;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bwzfXH",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c0"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-htpNat",
                              "isStatic": false,
                              "lastClassName": "c1",
                              "rules": Array [
                                "
  max-width: 500px;

  margin: auto;
  padding: ",
                                [Function],
                                ";

  background: ",
                                [Function],
                                ";
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-htpNat",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="c1"
                        >
                          <styled.div>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-bxivhb",
                                    "isStatic": false,
                                    "lastClassName": "c2",
                                    "rules": Array [
                                      "
  font-size: ",
                                      [Function],
                                      ";
  font-weight: bold;
  text-align: center;
",
                                    ],
                                  },
                                  "displayName": "styled.div",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-bxivhb",
                                  "target": "div",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <div
                                className="c2"
                              >
                                general:appTitle
                              </div>
                            </StyledComponent>
                          </styled.div>
                          <styled.div>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-ifAKCX",
                                    "isStatic": false,
                                    "lastClassName": "c3",
                                    "rules": Array [
                                      "
  margin-top: ",
                                      [Function],
                                      ";
",
                                    ],
                                  },
                                  "displayName": "styled.div",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-ifAKCX",
                                  "target": "div",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <div
                                className="c3"
                              >
                                <default
                                  features={
                                    Array [
                                      Object {
                                        "value": "React - Main library",
                                      },
                                      Object {
                                        "value": "SSR - Server Side Rendering",
                                      },
                                      Object {
                                        "value": "Typescript - Type checking",
                                      },
                                      Object {
                                        "value": "Redux - Global store",
                                      },
                                      Object {
                                        "value": "Redux-Saga - Global store manager",
                                      },
                                      Object {
                                        "value": "i18n - Localisation",
                                      },
                                    ]
                                  }
                                >
                                  <MockComponent
                                    features={
                                      Array [
                                        Object {
                                          "value": "React - Main library",
                                        },
                                        Object {
                                          "value": "SSR - Server Side Rendering",
                                        },
                                        Object {
                                          "value": "Typescript - Type checking",
                                        },
                                        Object {
                                          "value": "Redux - Global store",
                                        },
                                        Object {
                                          "value": "Redux-Saga - Global store manager",
                                        },
                                        Object {
                                          "value": "i18n - Localisation",
                                        },
                                      ]
                                    }
                                    originalComponent="Features"
                                  />
                                </default>
                              </div>
                            </StyledComponent>
                          </styled.div>
                          <LoaderComponent
                            error=""
                            isInitialLoaded={true}
                            isLoading={false}
                            render={[Function]}
                          >
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-EHOje",
                                      "isStatic": false,
                                      "lastClassName": "c4",
                                      "rules": Array [
                                        "
  margin-top: ",
                                        [Function],
                                        ";
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-EHOje",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="c4"
                                >
                                  <default
                                    releases={
                                      Array [
                                        Object {
                                          "html_url": "https://github.com/nikitapavets/nkit/releases/tag/v1",
                                          "id": 1,
                                          "name": "Release v1",
                                          "tag_name": "v1",
                                        },
                                      ]
                                    }
                                  >
                                    <MockComponent
                                      originalComponent="Releases"
                                      releases={
                                        Array [
                                          Object {
                                            "html_url": "https://github.com/nikitapavets/nkit/releases/tag/v1",
                                            "id": 1,
                                            "name": "Release v1",
                                            "tag_name": "v1",
                                          },
                                        ]
                                      }
                                    />
                                  </default>
                                </div>
                              </StyledComponent>
                            </styled.div>
                          </LoaderComponent>
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
            </section>
          </StyledComponent>
        </styled.section>
      </HomeComponent>
    </ThemeProvider>
  </Provider>
</Component>
`;
